### PR TITLE
Fixes #15458: Adapt techniques to stop agent if it's not in audit and the audit is enforced

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/15458

### CODE REVIEW

File: techniques/system/common/1.0/promises.st [hunk 1]
```diff
@@ -235,9 +238,34 @@ bundle agent check_disable_agent
...
+# All promises in this bundle MUST have action=>immediate,
+# to avoid locking issues in case of several close runs.
+##########################################################
+bundle agent check_rudder_force_audit_agent
+{
+  classes:
+      "should_be_in_audit" expression => fileexists("${g.rudder_force_audit_agent}");
+
+      # Only define this class when we're ready to die - this is a special class name in "abortclasses"
+      "should_not_continue"
```
Comment (amousset) [inline techniques/system/common/1.0/promises.st]: could you use the new abort mechanism instead?

Comment (amousset) [reply techniques/system/common/1.0/promises.st]: Like in ```bundle agent rudder_check_agent_version   
{
  methods:  
    cfengine_3_1|cfengine_3_2|cfengine_3_3|cfengine_3_4|cfengine_3_5|cfengine_3_6|cfengine_3_7|cfengine_3_8|cfengine_3_9::
      "any" usebundle => _abort("unsupported_agent", "This agent is not compatible with its Rudder server, please upgrade");
}
```

Comment (amousset) [reply techniques/system/common/1.0/promises.st]: (it send a control report to the webapp before aborting, which may be useful in the future)

**Comment by ncharles** [comment]
Commit modified

File: techniques/system/common/1.0/promises.st (hunk 1 above)
Comment (ncharles) [reply techniques/system/common/1.0/promises.st]: i had no idea it even existed

**Pull request merged**